src/Piece/PieceForm/PieceFields/PieceFields.js

```diff
@@ -46,6 +45,15 @@ export const PieceFields = ({
   const isLocationRequired = includes(createInventoryValues[values.format], INVENTORY_RECORDS_TYPE.instanceAndHolding);
   const isDisplayToPublic = values.displayOnHolding;
 
+  const {
+    locations,
+    locationIds,
+    isFetching,
+  } = useHoldingsAndLocations({
+    instanceId,
+    tenantId: values.receivingTenantId,
+  });
```

Comment:
Since all the data for the form is loaded in the container, I suggest not spreading this logic across all the components, but passing the data as props, as it is implemented now. This will simplify unit testing as well.